### DISCUSSION TITLE
Current user info inside hook filter

### DISCUSSION THREAD

**Discussion by gunners81** [directus/directus#2791] [Archive]
I need to partially remove payload if not eq with current user role, how to get current user role inside hook filter ?

'item.read.projects' => function (\Directus\Hook\Payload $payload) {
            $projects = $payload->getData();
            $current_user_role_id = "11"; // it should be dynamic

            foreach($projects as $index => $project) {
                if($project["user_role"] !== $current_user_role_id) {
                    $payload->remove($index);
                }
            }

            return $payload;
        }

**Comment by gunners81** [comment]
finally ... :)

$container = \Directus\Application\Application::getInstance()->getContainer();
$acl = $container->get('acl');
$logged_id = $acl->getUserId();

**Reply by buckhornmktg** [reply]
This documentation is also helpful for others looking for this solution: 
[https://docs.directus.io/advanced/api/data.html#fetching-data-within-hooks-and-endpoints](https://docs.directus.io/advanced/api/data.html#fetching-data-within-hooks-and-endpoints)

**Comment by buckhornmktg** [comment]
What was your solution for getting the role id?

**Comment by gunners81** [comment]
you can do it like this

`$container = \Directus\Application\Application::getInstance()->getContainer();`
`$acl = $container->get('acl');`
  `if( !$acl->isPublic() ) { //check only if not Public`
                `if( !$acl->isAdmin() ) { //check only if not Administrator`
                    `if ($acl->hasRole(6)) { //check if role is Manager(6) in my case`
` // do something with the payload `
`}`

**Comment by buckhornmktg** [comment]
Thank you! I was missing this part: $acl->hasRole(6)
